### PR TITLE
check `sonata_admin.field_description`, somewhere is null;

### LINKED ISSUE #4997: Filter doctrine_orm_model_autocomplete is broken in release 3.32.0
### Environment
any

#### Sonata packages
```
sonata-project/admin-bundle              3.32.0 3.32.0 The missing Symfony Admin Generator
sonata-project/block-bundle              3.12.0 3.12.0 Symfony SonataBlockBundle
sonata-project/cache                     2.0.1  2.0.1  Cache library
sonata-project/core-bundle               3.9.1  3.9.1  Symfony SonataCoreBundle
sonata-project/datagrid-bundle           2.3.1  2.3.1  Symfony SonataDatagridBundle
sonata-project/doctrine-orm-admin-bundle 3.4.2  3.4.2  Symfony Sonata / Integrate Doctrine ORM into the SonataAdminBundle
sonata-project/exporter                  1.8.0  1.8.0  Lightweight Exporter library
sonata-project/intl-bundle               2.4.0  2.4.0  Symfony SonataIntlBundle
```

#### Symfony packages
```
symfony/monolog-bundle     v3.2.0 v3.2.0 Symfony MonologBundle
symfony/phpunit-bridge     v4.0.6 v4.0.6 Symfony PHPUnit Bridge
symfony/polyfill-apcu      v1.7.0 v1.7.0 Symfony polyfill backporting apcu_* functions to lower PHP versions
symfony/polyfill-intl-icu  v1.7.0 v1.7.0 Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-mbstring  v1.7.0 v1.7.0 Symfony polyfill for the Mbstring extension
symfony/polyfill-php56     v1.7.0 v1.7.0 Symfony polyfill backporting some PHP 5.6+ features to lower PHP versions
symfony/polyfill-php70     v1.7.0 v1.7.0 Symfony polyfill backporting some PHP 7.0+ features to lower PHP versions
symfony/polyfill-php72     v1.7.0 v1.7.0 Symfony polyfill backporting some PHP 7.2+ features to lower PHP versions
symfony/polyfill-util      v1.7.0 v1.7.0 Symfony utilities for portability of PHP codes
symfony/security-acl       v3.0.1 v3.0.1 Symfony Security Component - ACL (Access Control List)
symfony/swiftmailer-bundle v3.2.0 v3.2.0 Symfony SwiftmailerBundle
symfony/symfony            v3.4.6 v4.0.6 The Symfony PHP framework
```

#### PHP version
```
PHP 7.2.2 (cli) (built: Feb  1 2018 11:50:40) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.2, Copyright (c) 1999-2018, by Zend Technologies
    with Xdebug v2.6.0, Copyright (c) 2002-2018, by Derick Rethans
```

## Subject
I'm using doctrine_orm_model_autocomplete filters in this way:
`$datagridMapper->add('user', 'doctrine_orm_model_autocomplete', [], null, ['property' => 'email'])`
As it is written in the documentation: https://sonata-project.org/bundles/doctrine-orm-admin/3-x/doc/reference/filter_field_definition.html#doctrine-orm-model-autocomplete
In related UserAdmin there is datagrid filter on `email` property. 
After the release 3.32.0 list page is broken:
`Impossible to access an attribute ("hasAssociationAdmin") on a null variable.`
https://github.com/sonata-project/SonataAdminBundle/blob/3.32.0/src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig#L40
Because sonata_admin.field_description in my case is null. Previous check (3.31.1) was working:
`sonata_admin.field_description.associationadmin is defined`
Is it a bug or incorrect documentation for autocomplete filter (usage example is incorrect)?

## Steps to reproduce
In any admin class add filter for relation field:
`$datagridMapper->add('user', 'doctrine_orm_model_autocomplete', [], null, ['property' => 'email'])`

## Expected results
List page is working

## Actual results
Twig_Error_Runtime
HTTP 500 Internal Server Error
Impossible to access an attribute ("hasAssociationAdmin") on a null variable.

### PR DESCRIPTION
**Pull request by qbbr**
Closes #4997

## Changelog

```markdown
### Fixed
- FieldDescription null check missing in ModelAutocompleteFilter
```

## Subject
fix "Impossible to access an attribute ("hasAssociationAdmin") on a null variable",
when using `ModelAutocompleteFilter` on configureDatagridFilters;

### CODE REVIEW

File: src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig [hunk 1]
```diff
@@ -37,7 +37,7 @@ file that was distributed with this source code.
         {% endif -%}
     </div>
 
-    {% if sonata_admin.field_description.hasAssociationAdmin %}
+    {% if sonata_admin.field_description.hasAssociationAdmin is defined %}
```
Comment (kunicmarko20) [inline src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig]: Youu should add check for `sonata_admin.field_description` and revert this change

**Comment by kunicmarko20** [comment]
Hey there!

It looks like you did not fill the pull request template properly. Maybe you
were confused, so let me walk you through this step by step.

# Explaining why you target this branch

> I am targeting this branch, because {reason}.

Here, you are supposed to replace `{reason}` with something like:

- "there is a BC-break" if you picked the master branch
- "there is no BC-break" if you picked the stable branch

# Searching and finding related issues

> Closes #{put_issue_number_here}

Please have a look at the list of issues and pull requests, and look for issues
that will be fixed, and pull requests that will become irrelevant should we
merge your PR. Each issue / pull request has a number shown right next to the
title. Kindly replace `{put_issue_number_here}` with whatever issue or PR is
relevant. Of course, you may specify several numbers.
On merge, [these issues and PRs will automatically get
closed](https://help.github.com/articles/closing-issues-using-keywords/).
If there is no issue or pull request that fits there, just remove the sentence
entirely.

# Filling the changelog

That one is really important to get right, because it will be used to craft a
release changelog such as [this
one](https://github.com/sonata-project/SonataAdminBundle/releases/tag/3.23.0).
The first thing you need to know about this is that if your pull request is
pedantic, meaning it does not fix a bug for the user, or adds a new feature, it
is not needed. This is typically the case for PRs that fix the build or add
some documentation.

3 important things:

1. Make sure to fill the changelog section with something directed at the end
user, that answers the question "What benefit do I get from upgrading?".
2. Make sure to remove empty sections.
3. Make sure to keep the markdown markup.

## Filling the TODO list

If you think your PR is ready, just remove it. If you have still some things
left to do, customize it.

## Filling the Subject

Last but not least, describe your PR as well as you can. Bear in mind that some
reviewers do not even use the library they are maintaining, so try to be
crystal clear.

**Comment by jordisala1991** [comment]
This would be the first place where we need this kind of check. Can we find why the field_description is null before adding this check?

**Comment by qbbr** [comment]
ok, I'll try to find out.

**Comment by jordisala1991** [comment]
Thank you @qbbr
